### PR TITLE
Refactoring node service with ref architecture recommendations

### PR DESCRIPTION
**Pull request by aalykiot**
Before we merge we should wait for @mhdawson 's feedback that everything is as it should be.

@cescoffier I'm also trying to build an advanced example with manual offset commits (that we discussed) but I'm experiencing some problems for now (I'll continue working on that so we will also have a more advanced example in the future).

### CODE REVIEW

File: barista-node/package.json
Comment (lholmquist) [inline barista-node/package.json]: Can we get this to be the same as the other nodeshift modules?  We are using semi-standard there

https://github.com/nodeshift/nodeshift/blob/master/package.json#L63
https://github.com/nodeshift/nodeshift/blob/master/.eslintrc.json

File: barista-node/package.json
Comment (lholmquist) [inline barista-node/package.json]: I would recommend against using `--fix` since in a CI environment,  you will never know that you had an issue, since it was fixed, but won't get commited back to the code

Comment (aalykiot) [reply barista-node/package.json]: Good point!

File: README.md
Comment (mhdawson) [inline README.md]: I understand the request to make it configurable, and include instructions in the doc of how to set it, but for the quarkus barista you don't have to set it in the default case, so I think it would be good if the Node.js had a default as well and you only need to set it if you were not using the localhost.

**Comment by mhdawson** [comment]
Overall looks good, just made one comment and agree with @lholmquist suggestions on the linting side as well.

**Review by mhdawson — APPROVED** [review]
LGTM once the existing comments are addressed.

File: README.md
Comment (mhdawson) [reply README.md]: Your latest change looks good (I've not tested it but assume you have)

**Comment by cescoffier** [comment]
@alexalikiotis should we merge this one? Or do you want to push more commit to it?

**Comment by aalykiot** [comment]
@cescoffier I think we should make some doc and folder-name changes according to @mhdawson before we merge it.

**Comment by aalykiot** [comment]
@cescoffier do you think we can rename the barista folders to `barista-java` and `barista-node` ?? (and update the docs accordingly)

**Comment by cescoffier** [comment]
Yes, it should be `barista-quarkus-kafka` and `barista-node-kafka` and `barista-quarkus-http`.

**Comment by aalykiot** [comment]
Great, I'll make the changes and commit them to this PR

**Comment by aalykiot** [comment]
@cescoffier So, I did make the folder name changes but some things from the JAVA side broke and the demo can't run.

After the rename, the quarkus side wasn't able event to build. Then I've made these changes to **pom.xml**
```
<modules>
    <module>barista-quarkus-http</module>
    <module>coffeeshop-service</module>
    <module>barista-quarkus-kafka</module>
</modules>
```

Now, it compiles fine but when I start the coffeeshop-service (seems to run well) I cannot access the http://localhost:8080

Should I just rename the folders, fix the node.js side and let you guys fix the JAVA side (including the k8s templates etc) ??

**Comment by cescoffier** [comment]
Yes, I can handle the JAva side. I'm no idea why this would not work... that picks my curiosity.

**Comment by aalykiot** [comment]
Cool, I think I haven't missed anything at the docs, but just to be sure @cescoffier take a look again.

**Review by cescoffier — APPROVED** [review]

**Comment by cescoffier** [comment]
@alexalikiotis I've pushed to your branch to get a green CI. 

I've tried the demo locally, it rocks! That's awesome! Seeing node.js and quarkus baristas collaborating is really cool!

**Comment by cescoffier** [comment]
@alexalikiotis When I did `npm install` it updated the `barista-node-kafka/package-lock.json` file. Should we commit it, or add it to .gitignore?

**Comment by aalykiot** [comment]
@cescoffier that's probably a mistake from my side. Let me fix it real quick.

**Comment by aalykiot** [comment]
@cescoffier done!

**Comment by cescoffier** [comment]
@alexalikiotis @mhdawson shall we merge? First outing planned on Friday :-)

**Comment by mhdawson** [comment]
@cescoffier sounds good to merge to me :)

**Comment by aalykiot** [comment]
@cescoffier are we gonna merge this one ??

**Comment by cescoffier** [comment]
Done!

**Pull request merged**
